pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -497,10 +518,39 @@ func (r *ReconcileUserSignup) provisionMasterUserRecord(userSignup *toolchainv1a
 		return r.wrapErrorWithStatusUpdate(logger, userSignup, r.setStatusFailedToCreateMUR, err,
 			"Error creating MasterUserRecord")
 	}
+	logger.Info("Created MasterUserRecord", "Name", mur.Name, "TargetCluster", targetCluster)
+	// increment the counter of MasterUserRecords
 	counter.IncrementMasterUserRecordCount()
+	// also, update the 'toolchain.dev.openshift.com/activation-counter' annotation value
+	return r.ensureActivationCounter(userSignup, logger)
+}
 
-	logger.Info("Created MasterUserRecord", "Name", mur.Name, "TargetCluster", targetCluster)
-	return nil
+// ensureActivationCounter increments the 'toolchain.dev.openshift.com/activation-counter' annotation value on the given UserSignup
+func (r *ReconcileUserSignup) ensureActivationCounter(userSignup *toolchainv1alpha1.UserSignup, logger logr.Logger) error {
+	// increment the counter of Returning Users
+	if activations, exists := userSignup.Annotations[toolchainv1alpha1.UserSignupActivationCounterAnnotationKey]; exists {
+		logger.Info("updating 'toolchain.dev.openshift.com/activation-counter' on active user")
+		if activations, err := strconv.Atoi(activations); err == nil {
+			// increment the value of the annotation
+			activations++
+			userSignup.Annotations[toolchainv1alpha1.UserSignupActivationCounterAnnotationKey] = strconv.Itoa(activations)
+			// increment the counter
+			counter.IncrementUsersPerActivationCounter(activations)
```

Comment:
shouldn't you decrement the counter for the previous number?

Reply:
ok, now I noticed that you do it as part of the function - could we change the name to:
```
IncrementAndDecrementUsersPerActivationCounter
```
or
```
UpdateUsersPerActivationCounters
```

Reply:
renamed to `UpdateUsersPerActivationCounters ` in e4db8fb